OpenFeature/src/main/java/dev/openfeature/sdk/FeatureProvider.kt

```diff
@@ -1,6 +1,8 @@
 package dev.openfeature.sdk
 
-interface FeatureProvider {
+import dev.openfeature.sdk.events.ProviderEventObserver
+
+interface FeatureProvider : ProviderEventObserver {
```

Comment:
I think we still miss an API for the Application Layer to be able to observe the Events. `FeatureProvider` is only meant to be consumed by the SDK.

As I hinted and mentioned in other comments in this PR, I think [Client](https://github.com/open-feature/kotlin-sdk/blob/c9eca676d0fd091a6146de2246435f9d52ab3cae/OpenFeature/src/main/java/dev/openfeature/sdk/Client.kt#L3) also needs to implement  the `ProviderEventObserver` interface

Reply:
after this change, each provider comes with EventObserver interface, it means we can observe each provider for the events. OpenFeatureAPI and client have access to the current provider and therefore we can observe events in the app layer as well. If we want to observe OpenFeatureAPI directly we can write a simple extension to just get the current provider which has been set and observe the events from that.

Reply:
the extension exists [here ](https://github.com/open-feature/kotlin-sdk/blob/69379d21ea10661d4111aaf4fbe0e02add99c9c4/OpenFeature/src/main/java/dev/openfeature/sdk/async/Extensions.kt#L38C2-L38C2) actually.

Reply:
I think we will need another extension to allow the application layer to consume _any_ event from the SDK. The application should not consumer events from the Provider directly. This can be addressed in a future PR though imo

Reply:
i'll add it in the same PR. after unplugged ;) :D